packages/@aws-cdk/aws-stepfunctions/lib/states/task-base.ts

```diff
@@ -347,4 +373,16 @@ export enum IntegrationPattern {
    * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-to-resource.html#connect-wait-token
    */
   WAIT_FOR_TASK_TOKEN = 'WAIT_FOR_TASK_TOKEN'
+}
+
+/**
+ * Specifies a target role assumed by the State Machine's execution role for invoking the task's resource.
+ */
+export interface Credentials {
+
+  /**
+   * The ARN of the IAM role to be assumed.
+   * Either a fixed value or a JSONPath expression can be used.
+   */
+  readonly roleArn: string;
```

Comment:
This can be much nicer if there was a concrete type denoting json path e.g.
```
role: iam.IRole | JsonPath;
```

Reply:
We wouldn't want to add this depth to these props just to get a roleArn. Typically, we pass IRole, but I see what you're trying to do here so I suggest you take the example of [Schedule](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-events/lib/schedule.ts) and do something like what they did there with an enum like class.

Reply:
I am not sure I 100% follow
1. When you say "this depth", do you mean the `iam.IRole | JsonPath` bit, or do you mean the hierarchy of of the prop `Credentials.roleArn`? If hierarchy, i.e. whether roleArn should be top level or not, the only reason I didn't do that, is I thought maybe in the future their can be more attributes in the Credentials object e.g. session name, etc. 
2. The role - rather than Credentials - is what really might benefit from different modelling similar to `Schedule`.

If we go for modelling a `Role` class here it would be something along the lines of
```ts
class Role {

public static jsonExpression(expression: string): Role

public static role(role: iam.Role): Role

```

The other thing is, I don't think there's a strong type for json expressions (to the best of my knowledge)

Reply:
I introduced a new change to introduce `TaskRole` (inspired by `Schedule` as you suggested) at https://github.com/aws/aws-cdk/pull/23012/commits/86b9a5b21101447b3390e367ad5354de117aa40b